garden-backend-service/tests/api/routes/modal/test_invocations.py

```diff
@@ -81,3 +82,72 @@ async def test_invoke_modal_fn(
         timeout=None, clear_on_success=True
     )
     assert mock_retry.call_count == 1
+
+
+@pytest.mark.asyncio
+@pytest.mark.integration
+async def test_invoke_modal_fn_rejects_request_if_user_is_over_usage_limit(
+    override_modal_vip,
+    client,
+    mock_db_session,
+    override_authenticated_dependency,
+    override_get_settings_dependency,
+    override_get_modal_client_dependency,
+    mocker,
+    mock_modal_app_create_request_one_function,
+    override_sandboxed_functions,
+):
+    # first, "deploy" a modal function to invoke
+    response = await client.post(
+        "/modal-apps", json=mock_modal_app_create_request_one_function
+    )
+    assert response.status_code == 200
+    response_data = response.json()
+    test_function_id = response_data["modal_function_ids"][0]
+
+    mock_function = MagicMock()
+    mock_function._invocation_function_id.return_value = "mock_function_id"
+    mock_invocation = AsyncMock()
+    mock_invocation.pop_function_call_outputs.return_value = MagicMock(
+        outputs=[
+            api_pb2.FunctionGetOutputsItem(
+                result=api_pb2.GenericResult(status=0, data=b"mock_result_data"),
+                data_format=api_pb2.DATA_FORMAT_PICKLE,
+            )
+        ]
+    )
+
+    mocker.patch("modal.functions._Function.lookup", return_value=mock_function)
+    mocker.patch("modal.functions._Invocation", return_value=mock_invocation)
+
+    # Mock retry_transient_errors to avoid outbound network calls
+    mock_retry = mocker.patch("src.api.routes.modal.invocations.retry_transient_errors")
+    mock_retry.side_effect = [
+        MagicMock(
+            function_call_id="mock_call_id", pipelined_inputs=["mock_input"]
+        ),  # FunctionMap response
+        MagicMock(
+            function_call_id="mock_call_id", pipelined_inputs=["mock_input"]
+        ),  # FunctionMap response
+    ]
```

Comment:
Yeah, I think it is because we are using the `side_effect` parameter instead of `return_value`. It threw an error on the second `POST` request without the second side effect.